packages/protocol/test/verifiers/RiscZeroVerifier.t.sol

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import "../L1/TaikoL1TestBase.sol";
+
+contract MockRiscZeroRemoteVerifier {
+    // To simulate failing and succeeding
+    bool public verifying;
+
+    function setVerifier(bool _verifying) public {
```

Comment:
If you want a contract to return a given value in test, please use https://book.getfoundry.sh/cheatcodes/mock-call.

Reply:
It will not work.

![kép](https://github.com/taikoxyz/taiko-mono/assets/51912515/4b9efaaa-554c-4609-a73c-5097b8bdfcfa)


Nr.1 (Red circle) it works because i directly "inject" false value as you suggested into the contract.

But Nr.2. (purple circle) it is not a direct "inject" (because `rv` callse `riscZeroRemoteVerifier`) it loses effect.